lib/id3c/api/routes.py

```diff
@@ -145,6 +146,99 @@ def receive_fhir(*, session):
     return "", 204
 
 
+@api_v1.route("/warehouse/identifier/<id>", methods = ['GET'])
+@authenticated_datastore_session_required
+def get_identifier(id, *, session):
+    """
+    Retrieve an identifier's metadata.
+
+    GET /warehouse/identifier/*id* to receive a JSON object containing the
+    identifier's record.  *id* may be a full UUID or shortened barcode.
+    """
+    LOG.debug(f"Fetching identifier «{id}»")
+
+    identifier = datastore.fetch_identifier(session, id)
+
+    return jsonify(identifier._asdict())
+
+
+@api_v1.route("/warehouse/identifier-sets", methods = ['GET'])
+@authenticated_datastore_session_required
+def get_identifier_sets(*, session):
+    """
+    Retrieve metadata about all identifier sets.
+
+    GET /warehouse/identifier-set to receive a JSON array of objects, each
+    containing a set's metadata fields.
+    """
+    LOG.debug(f"Fetching identifier sets")
+
+    sets = datastore.fetch_identifier_sets(session)
+
+    return jsonify([ set._asdict() for set in sets ])
+
+
+@api_v1.route("/warehouse/identifier-sets/<name>", methods = ['GET'])
+@authenticated_datastore_session_required
+def get_identifier_set(name, *, session):
+    """
+    Retrieve an identifier set's metadata.
+
+    GET /warehouse/identifier-set/*name* to receive a JSON object containing the set's
+    metadata fields.
+    """
+    LOG.debug(f"Fetching identifier set «{name}»")
+
+    set = datastore.fetch_identifier_set(session, name)
+
+    return jsonify(set._asdict())
+
+
+@api_v1.route("/warehouse/identifier-sets/<name>", methods = ['PUT'])
```

Comment:
Is there a reason you chose PUT instead of POST method here? Also, would it be beneficial to also be able to set the description value for the new record?

Reply:
Good question. In my experience:

PUT is usually used for idempotent operations acting directly on a specific resource (e.g. one with a unique id, in this case `<name>`).

POST is usually used for non-idempotent operations acting on a collection resource, to take arbitrary action (e.g. launch a job) or create new records where you don't know the unique id in advance (e.g. auto-numbered records). Responses to POSTs are often 302 Redirects to the newly-created specific resource.

There are lots of discussions about this, of course. One reasonable example is https://restfulapi.net/rest-put-vs-post/.

It would indeed probably be beneficial to allow a description to be set. I didn't think it necessary as a first pass (and it wasn't in my old wip), but it wouldn't be hard to add. I'll add it.

Reply:
Ok, that explanation makes sense. I came across that same page, and the last entry in the comparison table was throwing me off:
Generally, in practice, always use PUT for UPDATE operations. | Always use POST for CREATE operations.
-- | --

But your explanation matches up well with [this answer](https://stackoverflow.com/questions/630453/what-is-the-difference-between-post-and-put-in-http) on SO, so thanks for clearing that up for me!

Reply:
Added support for "description" with additional commits.